model/src/main/java/com/plhal/ares/model/DataRepositoryImpl.java

```diff
@@ -111,65 +92,83 @@ private void pridejClenyStatutarnihoOrganu() {
-                        tempSO.getPrijmeni().substring(0, 1).toUpperCase() + tempSO.getPrijmeni().substring(1));
+                jmeno = jmeno.substring(0, 1).toUpperCase() + jmeno.substring(1);
+                prijmeni = prijmeni.substring(0, 1).toUpperCase() + prijmeni.substring(1);
+
+                tempSO = StatutarniOrgan.builder()
+                        .prijmeni(prijmeni)
+                        .jmeno(jmeno)
+                        .funkce(funkce)
+                        .build();
 
                 // Přidávám člena statutárního orgánu do firmy
-                firma.getClenoveStatutarnihoOrganu().add(tempSO);
+                listStatutarniOrgan.add(tempSO);
             }
 
         }
 
+        return listStatutarniOrgan;
     }
 
     // Získávám údaje o základním kapitálu společnosti
-    private void zjistiZakladniKapital() {
+    private String zjistiZakladniKapital(Document doc) {
 
-        tempNodeList = doc.getElementsByTagName("dtt:Kapital");
+
+        String zakladniKapital;
+
+        NodeList tempNodeList = doc.getElementsByTagName("dtt:Kapital");
 
         // Zjišťuji, zda má firma základní kapitál
         if (tempNodeList.item(0) == null) {
-            firma.setZakladniKapital("Tato společnost nemá základní kapitál");
+            zakladniKapital = "Tato společnost nemá základní kapitál";
         } else {
-            tempE = (Element) tempNodeList.item(0);
+            Element tempE = (Element) tempNodeList.item(0);
 
             // Vhodně formátuji hodnotu o základním kapitálu (Odstraňuji zbytečný znak a
             // přidávám "Kč" nakonec)
-            String vklad = tempE.getElementsByTagName("dtt:Kc").item(0).getTextContent();
-            if (vklad.contains(";")) {
-                vklad = vklad.substring(0, vklad.indexOf(";"));
+            zakladniKapital = tempE.getElementsByTagName("dtt:Kc").item(0).getTextContent();
+            if (zakladniKapital.contains(";")) {
+                zakladniKapital = zakladniKapital.substring(0, zakladniKapital.indexOf(";"));
             }
-            vklad = vklad.concat(" Kč");
-            firma.setZakladniKapital(vklad);
-        }
+            zakladniKapital = zakladniKapital.concat(" Kč");
 
+        }
+        return zakladniKapital;
     }
 
     // Získávám údaje o názvu společnosti
-    private void zjistiNazev() {
-        tempNodeList = doc.getElementsByTagName("dtt:Obchodni_firma");
-        firma.setNazevFirmy(tempNodeList.item(0).getTextContent());
+    private String zjistiNazev(Document doc) {
+
+        String nazev;
+
+        NodeList tempNodeList = doc.getElementsByTagName("dtt:Obchodni_firma");
+        nazev = tempNodeList.item(0).getTextContent();
+
+        return nazev;
 
     }
 
     // Získávám údaje o sídle společnosti
-    private void zjistiSidlo() {
-        tempNodeList = doc.getElementsByTagName("dtt:Sidlo");
-        tempE = (Element) tempNodeList.item(0);
+    private String zjistiSidlo(Document doc) {
```

Comment:
Tady muzete take pouzit @NonNull. Na navratovem typu (zvazit, zda neni nullable - ale v navaznosti na datovou tridu si to nemyslim) i parametru.